frame/assets/src/tests/traits.rs

```diff
@@ -0,0 +1,662 @@
+			   prop_assert_ok!(<Pallet::<Test> as MultiCurrency<AssetId>>::deposit(asset_id, &account_2, first));
+			   prop_assert_ok!(<Pallet::<Test> as MultiReservableCurrency<AssetId>>::reserve(asset_id, &account_1, first + second ));
+			   //repatriate to free balance
+			   let repatriate_free = <Pallet::<Test> as MultiReservableCurrency<AssetId>>::repatriate_reserved(asset_id, &account_1, &account_2, second, BalanceStatus::Free).unwrap();
+			   prop_assert_eq!(<Pallet::<Test> as MultiCurrency<AssetId>>::free_balance(asset_id, &account_2), first + (second - repatriate_free));
+
+			   prop_assert_eq!(<Pallet::<Test> as MultiReservableCurrency<AssetId>>::reserved_balance(asset_id, &account_1), first + repatriate_free );
+
+
+				Ok(())
+			}).unwrap();
+		}
+
+
+		#[test]
+		fn test_unreserve_implementation(
+			account_1 in accounts(),
+			 asset_id in asset(),
+			(first, second, third) in valid_amounts_without_overflow_3()
+		) {
+			new_test_ext().execute_with(|| {
+
+				prop_assert_ok!(<Pallet::<Test> as MultiCurrency<AssetId>>::deposit(asset_id, &account_1, first + second + third));
+				prop_assert_eq!(<Pallet::<Test> as MultiCurrency<AssetId>>::free_balance(asset_id, &account_1), first + second + third);
+				prop_assert_ok!(<Pallet::<Test> as MultiReservableCurrency<AssetId>>::reserve(asset_id,&account_1, first+second+third));
+				prop_assert_eq!(<Pallet::<Test> as MultiReservableCurrency<AssetId>>::reserved_balance(asset_id, &account_1), first + second + third);
+				prop_assert_eq!(<Pallet::<Test> as MultiCurrency<AssetId>>::free_balance(asset_id, &account_1), 0);
+				//repatriate to free balance
+				let mut remaining = <Pallet::<Test> as MultiReservableCurrency<AssetId>>::unreserve(asset_id, &account_1, third);
+				prop_assert_eq!(<Pallet::<Test> as MultiCurrency<AssetId>>::free_balance(asset_id, &account_1), third - remaining);
+				let mut free_balance = <Pallet::<Test> as MultiCurrency<AssetId>>::free_balance(asset_id, &account_1);
+				remaining = <Pallet::<Test> as MultiReservableCurrency<AssetId>>::unreserve(asset_id, &account_1, second);
+				prop_assert_eq!(<Pallet::<Test> as MultiCurrency<AssetId>>::free_balance(asset_id, &account_1), free_balance + (second - remaining));
+
+				free_balance = <Pallet::<Test> as MultiCurrency<AccountId>>::free_balance(asset_id, &account_1);
+				remaining = <Pallet::<Test> as MultiReservableCurrency<AccountId>>::unreserve(asset_id, &account_1, first);
+				prop_assert_eq!(<Pallet::<Test> as MultiCurrency<AccountId>>::free_balance(asset_id, &account_1), free_balance + (first - remaining));
+				prop_assert_eq!(<Pallet::<Test> as MultiReservableCurrency<AccountId>>::reserved_balance(asset_id, &account_1), 0);
+				prop_assert_eq!(<Pallet::<Test> as MultiCurrency<AccountId>>::free_balance(asset_id, &account_1), first + second + third);
+
+				Ok(())
+			}).unwrap();
+		}
+	}
+}
+
+/*
+type PositiveImbalance: Imbalance<Self::Balance, Opposite = Self::NegativeImbalance>
```

Comment:
it is like if this comment should be moved closer to place where we should implement this? or it is TODO for tests? not clear